src/components/Image.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import React from "react";
+import { Image as RNImage } from "react-native";
+import { connect } from "react-redux";
+import type { Connector } from "react-redux";
+import type { ImageDetails } from "../data/image";
+import { getImageDetails as GetImageDetails } from "../data/image";
+import type { FieldRef } from "../data/field-ref";
+
+type OwnProps = {
+  reference: FieldRef
+};
+
+type StateProps = {
+  getImageDetails: string => ?ImageDetails
+};
+
+type Props = OwnProps & StateProps;
+
+export const Image = ({ reference, getImageDetails, ...props }: Props) => (
+  <RNImage source={getImageDetails(reference.sys.id)} {...props} />
+);
+
+// Note we must add a return type here for react-redux connect to work
+// with flow correctly. If not provided is silently fails if types do
+// not line up. See https://github.com/facebook/flow/issues/5343
+const mapStateToProps = (state, ownProps: OwnProps): Props => ({
+  getImageDetails: GetImageDetails(state.data.images),
```

Comment:
I don't like the naming of `GetImageDetails` (with a capital G). I first thought this might be a React component. Maybe we can import it as something like `createImageDetailsGetter`. What do you think?